code/game/objects/items/devices/spyglasses.dm

```diff
@@ -0,0 +1,81 @@
+//antag spyglasses. meant to be an example for map_popups.dm
+/obj/item/clothing/glasses/regular/spy
+	desc = "Made by Nerd. Co's infiltration and surveillance department. Upon closer inspection, there's a small screen in each lens."
+	var/obj/item/spy_bug/linked_bug
+
+/obj/item/clothing/glasses/regular/spy/proc/show_to_user(var/mob/user)//this is the meat of it. most of the map_popup usage is in this.
+	if(!user)
+		return
+	if(!user.client)
+		return	
```

Comment:
You can combine these
`if(!user?.client)`

Reply:
I guess, I generally don't like doing that because it's less readable, but if it's requested I'll do it.